lib/services/resource.go

```diff
@@ -638,3 +640,41 @@ func (u *UnknownResource) UnmarshalJSON(raw []byte) error {
 	copy(u.Raw, raw)
 	return nil
 }
+
+type bootstrapResourcesType struct {
+	LocalResources  []types.Resource
+	DeviceResources []types.DeviceV1
+}
+
+func FilterBootstrapResourcesByType(resources ...types.Resource) (filteredResources bootstrapResourcesType) {
+	for _, res := range resources {
+		switch r := res.(type) {
+		case *types.DeviceV1:
+			filteredResources.DeviceResources = append(filteredResources.DeviceResources, *r)
+		default:
+			filteredResources.LocalResources = append(filteredResources.LocalResources, r)
+		}
+	}
+	return
+}
+
+type BootstrapDeviceResource func(context.Context, backend.Backend, *types.DeviceV1) error
+
+var eDeviceBootstrapper BootstrapDeviceResource
```

Comment:
Drop the "e" prefix, seems like a needless assumption here.

```suggestion
var deviceBootstrapper BootstrapDeviceResource
```